argoverse/evaluation/eval_detection.py

```diff
@@ -0,0 +1,410 @@
+            ninst = cls_to_ninst[cls_name]
+            ranks = cls_stats[:, -1].argsort()[::-1]
+            cls_stats = cls_stats[ranks]
+
+            for i, _ in enumerate(self.detection_cfg.affinity_threshs):
+                tp = cls_stats[:, i].astype(bool)
+
+                cumulative_tp = np.cumsum(tp, dtype=np.int)
+                cumulative_fp = np.cumsum(~tp, dtype=np.int)
+                cumulative_fn = ninst - cumulative_tp
+
+                precisions = cumulative_tp / (cumulative_tp + cumulative_fp + np.finfo(float).eps)
+                recalls = cumulative_tp / (cumulative_tp + cumulative_fn)
+
+                precisions = interp(precisions)
+                precisions_interp = np.interp(recalls_interp, recalls, precisions, right=0)
+
+                ap_th = precisions_interp.mean()
+                summary[cls_name] += [ap_th]
+
+                if self.detection_cfg.save_figs:
+                    self.plot(recalls_interp, precisions_interp, cls_name)
+
+            # AP Metric
+            ap = np.array(summary[cls_name][:num_ths]).mean()
+
+            # Select only the true positives for each instance.
+            tp_metrics_mask = ~np.isnan(cls_stats[:, num_ths : num_ths + N_TP_ERRORS]).all(axis=1)
+
+            # If there are no true positives set tp errors to their maximum values due to normalization below)
+            if ~tp_metrics_mask.any():
+                tp_metrics = self.detection_cfg.tp_normalization_terms
+            else:
+                # Calculate TP metrics.
+                tp_metrics = np.mean(cls_stats[:, num_ths : num_ths + N_TP_ERRORS][tp_metrics_mask], axis=0)
+
+            # Convert errors to scores
+            tp_scores = 1 - (tp_metrics / self.detection_cfg.tp_normalization_terms)
+
+            # Compute Composite Detection Score (CDS)
+            cds = ap * tp_scores.mean()
+
+            summary[cls_name] = [ap, *tp_metrics, cds]
+
+        logger.info(f"summary = {summary}")
+        return summary
+
+    def plot(self, rec_interp: np.ndarray, prec_interp: np.ndarray, cls_name: str) -> None:
+        """Plot and save the precision recall curve.
+
+        Args:
+            rec_interp: Interpolated recall data of shape (N,).
+            prec_interp: Interpolated precision data of shape (N,).
+            cls_name: Class name.
+        """
+        plt.plot(rec_interp, prec_interp)
+        plt.title("PR Curve")
+        plt.xlabel("Recall")
+        plt.ylabel("Precision")
+        plt.savefig(f"{self.figures_fpath}/{cls_name}.png")
+        plt.close()
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument("-d", "--dt_fpath", type=str, help="Detection root folder path.", required=True)
+    parser.add_argument("-g", "--gt_fpath", type=str, help="Ground truth root folder path.", required=True)
+    parser.add_argument("-f", "--fig_fpath", type=str, help="Figures root folder path.", default="figs/")
```

Comment:
the default should just be `figs` i think, not `figs/`, since we add the `/` in the matplotlib section